### PR TITLE
New Relay RP version 2017-04-01

### PR DESCRIPTION
**Pull request by v-Ajnava**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-relay/2017-04-01/swagger/relay.json [hunk 1]
```diff
@@ -0,0 +1,2195 @@
...
+          },
+          {
+            "$ref": "#/parameters/namespaceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RelayNamespacePatch"
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: I noticed this model was updated as "Patch" just for the patch operation and doesn't match the model from PUT and GET operations, I see that the difference is "location" being a required property or not. Is that intentional?
@ravbhatnagar could you take a look at this case, since it's an RPC violation?https://travis-ci.org/Azure/azure-rest-api-specs/jobs/242668994#L1203

File: arm-relay/2017-04-01/swagger/relay.json [hunk 2]
```diff
@@ -0,0 +1,2195 @@
...
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Namespace successfuly updated.",
+            "schema": {
+              "$ref": "#/definitions/RelayNamespacePatch"
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: note that response model name is changing compared to the previous API version, is this intentional?

File: arm-relay/2017-04-01/swagger/relay.json [hunk 3]
```diff
@@ -0,0 +1,2195 @@
...
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful",
+            "schema": {
+              "$ref": "#/definitions/AccessKeys"
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: Please note that model name (class name in generated code) is changes compared to previous api-version, is this intentional?

File: arm-relay/2017-04-01/swagger/relay.json [hunk 4]
```diff
@@ -0,0 +1,2195 @@
...
+    "RelayNamespace": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "Sku of the Namespace."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "properties": {
+            "provisioningState": {
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: is there a reason why the $ref was removed in this api-version (I see there used to be a ref in the previous api-version), it doesn't make a difference from the swagger point of view, but if you're reusing the properties, it avoid repetition and copy/paste mistakes.

File: arm-relay/2017-04-01/swagger/relay.json [hunk 5]
```diff
@@ -0,0 +1,2195 @@
...
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of this Sku",
+          "enum": [
+            "Standard"
+          ],
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": false
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: I see the old api-version had modelAsString set to true, is this change intentional?

File: arm-relay/2017-04-01/swagger/relay.json [hunk 6]
```diff
@@ -0,0 +1,2195 @@
...
+        },
+        "tier": {
+          "type": "string",
+          "description": "The tier of this particular SKU",
+          "enum": [
+            "Standard"
+          ],
+          "x-ms-enum": {
+            "name": "SkuTier",
+            "modelAsString": false
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: similar question as above

File: arm-relay/2017-04-01/swagger/relay.json [hunk 7]
```diff
@@ -0,0 +1,2195 @@
...
+        }
+      },
+      "description": "The response of the List Namespace operation."
+    },
+    "AuthorizationRule": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "properties": {
+            "rights": {
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: is there a reason these properties are defined inline in this api-version but had a $ref in the previous api-version?

File: arm-relay/2017-04-01/swagger/relay.json [hunk 8]
```diff
@@ -0,0 +1,2195 @@
...
+              "items": {
+                "type": "string",
+                "enum": [
+                  "Manage",
+                  "Send",
+                  "Listen"
+                ],
+                "x-ms-enum": {
+                  "name": "AccessRights",
+                  "modelAsString": false
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: I see this value was set to true before, is the value false intentional?

File: arm-relay/2017-04-01/swagger/relay.json [hunk 9]
```diff
@@ -0,0 +1,2195 @@
...
+          "description": "A base64-encoded 256-bit primary key for signing and validating the SAS token"
+        },
+        "secondaryKey": {
+          "type": "string",
+          "description": "A base64-encoded 256-bit secondary key for signing and validating the SAS token"
+        },
+        "keyName": {
+          "type": "string",
+          "description": "A string that describes the authorization rule"
+
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: is this set to false intentionally?

**Comment by veronicagg** [comment]
@v-Ajnava It looks like this version is based on the older api version, if so it'd be great to have a commit of the older API version first in the new folder, so the review shows the differences. "When you are creating the swagger spec for the new api-version, please copy the swagger spec from the previous version in to the new api-versioned folder and commit it. After that overwrite it with the changes for the new api-version. This makes it easy for us to review the changes." (from https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md#filenames-and-folder-structure).

**Review by veronicagg — CHANGES_REQUESTED** [review]
I see only a small set of changes compared to the previous api-version, I've added some questions/comments inline.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 1 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Reverted the changes and moved the location required operations section as its only required for Namespace PUT.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 2 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Reverted the changes.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 3 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: yes, to have more meaningful name.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 4 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: 'provisioningState' will not be used any where else, so made it inline.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 7 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: 'rights' is been used only for authorization rules, so made it inline.

**Comment by salameer** [comment]
Hi @v-Ajnava,

please change the status to review required, after you address the changes, if not already done

File: arm-relay/2017-04-01/swagger/relay.json (hunk 8 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Yes, we want to expose it as an enum to people consuming the SDK so that they can strong type it

File: arm-relay/2017-04-01/swagger/relay.json (hunk 6 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Yes, we want to expose it as an enum to people consuming the SDK so that they can strong type it

File: arm-relay/2017-04-01/swagger/relay.json (hunk 5 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Yes, we want to expose it as an enum to people consuming the SDK so that they can strong type it

File: arm-relay/2017-04-01/swagger/relay.json (hunk 9 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Yes, we want to expose it as an enum to people consuming the SDK so that they can strong type it

**Comment by v-Ajnava** [comment]
@veronicagg. Created 3 commits, 1 with older version, 2 new version swagger and 3 with examples.

**Comment by v-Ajnava** [comment]
@salameer and @veronicagg, I am not able to change the labels. may be because of the cla-required tag.

File: arm-relay/2017-04-01/swagger/relay.json [hunk 10]
```diff
@@ -1575,9 +1581,6 @@
           "description": "Resource tags"
         }
       },
-      "required": [
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: Should the models that inherit from here be Tracked Resources? I'm asking because by removing the "required" location, they are not considered "tracked" - see definition of tracked resource at https://github.com/Azure/azure-rest-api-specs-pr/blob/master/common-types/resource-management/v1/types.json#L23

**Review by veronicagg — CHANGES_REQUESTED** [review]
one more question inline, rest looks good.
It appears that you need to sign CLA, could you check on that?

File: arm-relay/2017-04-01/swagger/relay.json (hunk 10 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: If we keep the Required in the tracked resource, than for the patch operation we cant use the same Model as the location is required. can you please suggest how to implement it so that we can use the same model for PUT and PATCH.

Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: that's the reason I was asking if they are really tracked resources, and if so, I pulled in @ravbhatnagar to comment as it has to do with the api design. @ravbhatnagar could you provide your thoughts on this issue?

Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Have added a new Namespace update parameters, Resource for namespace patch and added back location required to TrackedResource. Please review and let me know your views.

File: arm-relay/2017-04-01/swagger/relay.json [hunk 11]
```diff
@@ -1586,6 +1583,30 @@
           "$ref": "#/definitions/Resource"
         }
       ],
+	  "required": [
+		"location"
+	  ],
+      "description": "Definition of Resource"
+    },
+	"ResourceNamespacePatch": {
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: minor: please fix indenting

File: arm-relay/2017-04-01/swagger/relay.json [hunk 12]
```diff
@@ -1813,6 +1806,57 @@
         }
       ]
     },
+	"RelayNamespaceProperties": {
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: minor: please fix indenting

File: arm-relay/2017-04-01/swagger/relay.json [hunk 13]
```diff
@@ -1813,6 +1806,57 @@
...
+        },
+        "metricId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Identifier for Azure Insights metrics"
+        }
+      },
+      "description": "Properties of the Namespace."
+    },
+	"RelayUpdateParameters": {
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: minor: please fix indenting

**Review by veronicagg — CHANGES_REQUESTED** [review]
@v-Ajnava looks fine from my side, just a few minor comments on formatting of the spec. Do you need to sign CLA? 
@ravbhatnagar could you please sign-off from your side, particularly regarding this discussion https://github.com/Azure/azure-rest-api-specs/pull/1306#discussion_r122349530 
thanks!

File: arm-relay/2017-04-01/swagger/relay.json (hunk 13 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: fixed

File: arm-relay/2017-04-01/swagger/relay.json (hunk 12 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: fixed

File: arm-relay/2017-04-01/swagger/relay.json (hunk 11 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: fixed

**Comment by v-Ajnava** [comment]
previously I had signed CLA and yesterday also again i signed, can you please check it.

**Comment by veronicagg** [comment]
@v-Ajnava are you seeing any confirmation after signing CLA? I believe the bot is supposed to update the label once CLA is properly signed (see https://github.com/Azure/azure-rest-api-specs/pull/1306#issuecomment-309116820 as reference), but I'm not seeing it in this case.

**Comment by veronicagg** [comment]
@v-Ajnava Additionally, just to clarify, make sure that the spec actually matches the service, by submitting the PR and signing the CLA, you agree that the spec describes the service accurately. It may be a good idea to have someone else from your team also approve the PR.

File: arm-relay/2017-04-01/swagger/relay.json [hunk 14]
```diff
@@ -0,0 +1,2171 @@
...
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Relay/namespaces/{namespaceName}/AuthorizationRules/{authorizationRuleName}": {
```
Comment (ravbhatnagar) [inline arm-relay/2017-04-01/swagger/relay.json]: Couldn't authorizationRules be a property of the namespaces resource instead of being a child resource of namespaces?

Comment (ravbhatnagar) [inline arm-relay/2017-04-01/swagger/relay.json]: Wont that make your model more simplified - where during create of a namespace or a hybridConnection, you can specify the authorizationRules for that entity. What am I missing?

File: arm-relay/2017-04-01/swagger/relay.json [hunk 15]
```diff
@@ -0,0 +1,2171 @@
...
+          "default": {
+            "description": "Relay error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Relay/namespaces/{namespaceName}/AuthorizationRules/{authorizationRuleName}/regenerateKeys": {
```
Comment (ravbhatnagar) [inline arm-relay/2017-04-01/swagger/relay.json]: Can you please help me understand why is the regenerateKeys or listKeys in context of authorizationRules? I believe the key will be different based on the value of the AuthorizationRule (Manage, Send, Listen). Is that correct?
Is the scenario that for a particular namespace, I can create 3 different authorizationRules (with the different allowed values)?

File: arm-relay/2017-04-01/swagger/relay.json [hunk 16]
```diff
@@ -0,0 +1,2171 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Namespace create request succeeded",
+            "schema": {
+              "$ref": "#/definitions/RelayNamespace"
+            }
+          },
+          "202": {
```
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Resolved.

File: arm-relay/2017-04-01/swagger/relay.json [hunk 17]
```diff
@@ -0,0 +1,2171 @@
...
+          "200": {
+            "description": "check availability returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/CheckNameAvailabilityResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Relay/Namespaces": {
```
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Resolved.

File: arm-relay/2017-04-01/swagger/relay.json [hunk 18]
```diff
@@ -0,0 +1,2171 @@
...
+          "200": {
+            "description": "Namespace successfuly updated.",
+            "schema": {
+              "$ref": "#/definitions/RelayNamespace"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Relay/namespaces/{namespaceName}/AuthorizationRules": {
```
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Resolved

File: arm-relay/2017-04-01/swagger/relay.json [hunk 19]
```diff
@@ -0,0 +1,2171 @@
...
+        }
+      ],
+      "required": [
+        "location"
+      ],
+      "description": "Definition of Resource"
+    },
+    "ResourceNamespacePatch": {
+      "properties": {
+        "location": {
```
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Removed location, Resolved

File: arm-relay/2017-04-01/swagger/relay.json [hunk 20]
```diff
@@ -0,0 +1,2171 @@
...
+          "format": "date-time",
+          "readOnly": true,
+          "description": "The time the namespace was updated."
+        },
+        "serviceBusEndpoint": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint you can use to perform Service Bus operations."
+        },
+        "metricId": {
```
Comment (dsouzaarun) [inline arm-relay/2017-04-01/swagger/relay.json]: Insights RP (shoebox) has a feature that allows mapping of InternalID to ARM ID,  Which is what we use because the id the service uses to emit metrics is not a fully qualified ARM ID.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 14 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Arun replied on email.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 15 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Arun - dsouzaarun, replied on email.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 20 above)
Comment (ravbhatnagar) [inline arm-relay/2017-04-01/swagger/relay.json]: Got it!

File: arm-relay/2017-04-01/swagger/relay.json (hunk 15 above)
Comment (ravbhatnagar) [inline arm-relay/2017-04-01/swagger/relay.json]: Yes, resolved over mail

File: arm-relay/2017-04-01/swagger/relay.json (hunk 14 above)
Comment (ravbhatnagar) [inline arm-relay/2017-04-01/swagger/relay.json]: Yes, resolved on email.

File: arm-relay/2017-04-01/swagger/relay.json [hunk 21]
```diff
@@ -0,0 +1,2203 @@
...
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Relay/CheckNameAvailability": {
```
Comment (ravbhatnagar) [inline arm-relay/2017-04-01/swagger/relay.json]: Please fix camelcase

File: arm-relay/2017-04-01/swagger/relay.json [hunk 22]
```diff
@@ -0,0 +1,2203 @@
...
+      "description": "Description of a Namespace resource.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "RelayNamespaceProperties": {
+      "properties": {
+        "provisioningState": {
```
Comment (ravbhatnagar) [inline arm-relay/2017-04-01/swagger/relay.json]: PLease include the supported values for provisioningState. It must support at least the 3 provisioning states - Succeeded, failed and cancelled.

**Review by ravbhatnagar — COMMENTED** [review]
A couple of minor camel case things remain but looks fine otherwise.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 22 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: added supporting values in description

File: arm-relay/2017-04-01/swagger/relay.json (hunk 21 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: fixed.

**Comment by v-Ajnava** [comment]
@veronicagg, the CLA issue is resolved.

File: arm-relay/2017-04-01/swagger/relay.json [hunk 23]
```diff
@@ -227,8 +233,11 @@
               "$ref": "#/definitions/RelayNamespace"
             }
           },
-          "202": {
-            "description": "Namespace create request accepted"
+          "201": {
+            "description": "Namespace create request accepted",
+			"schema": {
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: indenting seems off

File: arm-relay/2017-04-01/swagger/relay.json [hunk 24]
```diff
@@ -361,11 +370,23 @@
             "schema": {
               "$ref": "#/definitions/RelayNamespace"
             }
+          },
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: is PATCH a long running operation, if so, please use "x-ms-long-running-operation: true"

File: arm-relay/2017-04-01/swagger/relay.json [hunk 25]
```diff
@@ -1568,13 +1595,22 @@
       "properties": {
         "location": {
           "type": "string",
+		  "x-ms-mutability": [
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: indentation is off

File: arm-relay/2017-04-01/swagger/relay.json [hunk 26]
```diff
@@ -1568,13 +1595,22 @@
...
+            "create"
+          ],
           "description": "Resource location"
         },
         "tags": {
           "type": "object",
           "additionalProperties": {
             "type": "string"
           },
+          "x-ms-mutability": [
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: is there a reason why you included this, please see: "When the extension is applied with all the three values; "x-ms-mutability": ["create", "read", "update"] (order of the values is not important) OR when this extension is not applied on a model property; it has the same effect in both the cases. Thus applying this extension with all the three values on all the settable properties is not required. This will ensure the spec is visibly cleaner" (from https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#rules )

File: arm-relay/2017-04-01/swagger/relay.json (hunk 22 above)
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: should it be an an enum, with modelAsString:true ? see https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum

**Review by veronicagg — CHANGES_REQUESTED** [review]
Please fix examples per https://travis-ci.org/Azure/azure-rest-api-specs/jobs/245150829#L1170
A lot of the changes made to the spec, are service impacting changes, does the spec reflect the actual service? is the service deployed to ARM as described in this spec? 
To approve/merge this spec, the service needs to be deployed and matching the spec, since after the spec is merged an SDK can be generated and needs to be in a working state, for customer usage. Please let me know.
Minor: there are some empty lines in the document, please remove them if possible.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 25 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: resolved.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 23 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: resolved.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 26 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: resolved. removed the x-ms-mutability for tags.

**Comment by v-Ajnava** [comment]
@veronicagg, yes the spec matches the deployed to ARM.  I have generated the SDK and verified it works with new RP version.

**Comment by v-Ajnava** [comment]
@veronicagg, my bad, I added long running to patch in previous commit. reverted it. also added 'updating' to provisioning state enum.

File: arm-relay/2017-04-01/swagger/relay.json [hunk 27]
```diff
@@ -1842,8 +1837,18 @@
...
+            "Created",
+            "Succeeded",
+            "Deleted",
+            "Failed",
+            "Updating",
+            "Unknown"
+          ],
+          "x-ms-enum": {
+            "name": "provisioningStateEnum",
+            "modelAsString": false
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: should this be true? Making it "false" will make this an enum, if there are other possible values supported by the service now or later, this may fail/cause breaking changes in the SDK. Could you confirm you want to make this a real enum?

File: arm-relay/2017-04-01/swagger/relay.json [hunk 28]
```diff
@@ -1842,8 +1837,18 @@
       "properties": {
         "provisioningState": {
           "type": "string",
-          "readOnly": true,
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: should it not be readonly?

File: arm-relay/2017-04-01/swagger/relay.json [hunk 29]
```diff
@@ -383,8 +383,7 @@
               "$ref": "#/definitions/ErrorResponse"
             }
           }
-        },
-        "x-ms-long-running-operation": true
```
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: is this one not a long running operation?

**Review by veronicagg — CHANGES_REQUESTED** [review]
@v-Ajnava I'm a little confused, is PATCH not a long running operation? Please make sure that any long running operation is marked with the tag, for proper SDK generation. This should be determined by how the service behaves and the status codes returned by the operation.

**Comment by veronicagg** [comment]
@v-Ajnava any updates?

File: arm-relay/2017-04-01/swagger/relay.json (hunk 28 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: provisioningState is readonly, confirmed with developer.

File: arm-relay/2017-04-01/swagger/relay.json (hunk 29 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: only the delete operation is long running. confirmed with developers

File: arm-relay/2017-04-01/swagger/relay.json (hunk 27 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: Yes, we are using enum. confirmed with developers

File: arm-relay/2017-04-01/swagger/relay.json (hunk 28 above)
Comment (veronicagg) [inline arm-relay/2017-04-01/swagger/relay.json]: ok, thanks for the follow up, whenever you push the update, I'll take a look.

**Comment by veronicagg** [comment]
@ravbhatnagar could you take a look at the updates and approve/request changes? thanks!

**Comment by ravbhatnagar** [comment]
Looks good. I thought I already ACK'd and approved this earlier. Apologies for the delay. Thanks

**Comment by veronicagg** [comment]
@v-Ajnava I believe there's a readonly update pending, from the comments above, once you publish that, I can approve the PR. Thanks!

**Comment by v-Ajnava** [comment]
@veronicagg, the Travis CI build failed with 'E: Failed to fetch https://debian.neo4j.org/repo/stable/Packages.gz  Hash Sum mismatch', is this transient error ?

**Comment by veronicagg** [comment]
@v-Ajnava we're looking into CI failures, I'll restart the job once that gets fixed.

**Review by veronicagg — APPROVED** [review]

File: arm-relay/2017-04-01/swagger/relay.json (hunk 16 above)
Comment (balajikris) [inline arm-relay/2017-04-01/swagger/relay.json]: Sorry to pick on a old thread, but I'm confused here. This thread says it was resolved, but in latest master I still see a 202 and not a 201 + ProvisioningState as @ravbhatnagar suggested. 
Am I missing something? Was this resolved "Fixed" or resolved as "won't fix" ?

File: arm-relay/2017-04-01/swagger/relay.json (hunk 18 above)
Comment (balajikris) [inline arm-relay/2017-04-01/swagger/relay.json]: Same here. By resolved, did you mean "resolved as wont-fix" ? @v-Ajnava

Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: @balajikris , I pulled the latest 2017-04-01 relay swagger, these comments are resolved with suggested changes. can you please verify it again or let me know if i am missing something. 
Link: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-relay/2017-04-01/swagger/relay.json

File: arm-relay/2017-04-01/swagger/relay.json (hunk 16 above)
Comment (v-Ajnava) [inline arm-relay/2017-04-01/swagger/relay.json]: @balajikris , I pulled the latest 2017-04-01 relay swagger, these comments are resolved with suggested changes. can you please verify it again or let me know if i am missing something.
Link: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-relay/2017-04-01/swagger/relay.json
